### PR TITLE
Use an Artifact ID rather than a composite OR statment in several places.

### PR DESCRIPTION
**Pull request by alanmcruickshank**
In several places we had:

```sql
on models.command_invocation_id = latest_compile.command_invocation_id
    or models.dbt_cloud_run_id = latest_compile.dbt_cloud_run_id
```

This PR replaces that idiom with a single identifier created in the base artifacts model, made from a coalesce of the two original IDs, then hashed. I think it's much cleaner and more future proof.

Would appreciate feedback on whether `SHA2` is overkill or whether `MD5` might be more pragmatic @NiallRees.

### CODE REVIEW

File: models/staging/stg_dbt__artifacts.sql [hunk 1]
```diff
@@ -36,6 +36,8 @@ artifacts as (
     select
         command_invocation_id,
         dbt_cloud_run_id,
+        -- This ID provides a reliable ID, regardless of whether running in a local or cloud environment.
+        sha2_hex(coalesce(dbt_cloud_run_id, command_invocation_id), 256) as artifact_run_id,
```
Comment (alanmcruickshank) [inline models/staging/stg_dbt__artifacts.sql]: @NiallRees - this is the bit that I'd love your view on.

Comment (NiallRees) [reply models/staging/stg_dbt__artifacts.sql]: Great idea. Linked issue: https://github.com/brooklyn-data/dbt_artifacts/issues/53

File: models/dim_dbt__current_models.sql [hunk 2]
```diff
@@ -22,9 +22,7 @@ model_executions as (
 -- Get the most recent comile run
 latest_compile as (
 
-    select
-        command_invocation_id,
-        dbt_cloud_run_id
+    select artifact_run_id
```
Comment (NiallRees) [inline models/dim_dbt__current_models.sql:25]: Even if we're joining on this new key, can we keep the `command_invocation_id` and `dbt_cloud_run_id` in these models? Both are potentially useful to users.

Comment (NiallRees) [reply models/dim_dbt__current_models.sql:25]: Oh nvm they weren't in the model output anyway.

**Review by NiallRees — APPROVED** [review]

**Pull request merged**